server_ftp/lib/abstract_ftp_server.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from abc import ABC, abstractmethod
+
+
+class AbstractFtpServer(ABC):
+    def __init__(self, url, user, password, port, path):
+        self.server = self.connect(url, user, password, port, path)
+        super().__init__()
+
+    def get_server(self):
+        return self.server
+
+    @abstractmethod
+    def connect(self, url, user, password, port, path):
+        """ Connect to object """
+        raise NotImplementedError
+
+    @abstractmethod
+    def close(self):
+        """ Close connection """
+        raise NotImplementedError
+
+    @abstractmethod
+    def put(self, file_like, name):
```

Comment:
As put is not used in FTP but is used for storing local files mentioned by name in pysftp, this is confusing for people knowing pysftp. Stick to putfo() for this one as well. By the way, do not forget the getfo() and its implementations.